packages/datetime/src/dateRangeInput.tsx

```diff
@@ -735,6 +736,38 @@ export class DateRangeInput extends AbstractPureComponent<IDateRangeInputProps,
         return isFocused && inputRef !== undefined && document.activeElement !== inputRef;
     }
 
+    private getIsOpenValueWhenDateChanges = (nextSelectedStart: Date, nextSelectedEnd: Date): boolean => {
+        if (this.props.closeOnSelection) {
+            // trivial case when TimePicker is not shown
+            if (this.props.timePrecision == null) {
+                return false;
+            }
+
+            let currentStartDate: Date = new Date(new Date().setHours(0, 0, 0, 0));
+            let currentEndDate: Date = new Date(new Date().setHours(0, 0, 0, 0));
+
+            if (this.isControlled()) {
```

Comment:
I think we could DRY this up a little with a helper function... it's repeated on L786. we could have something like

```
private getSelectedRange(fallbackRange: [Date, Date]) {
    if (this.isControlled()) {
        ...
    } else {
        ...
    }
}
```

and the "fallback" would be what you have on L746-747

Reply:
Modified existing `getSelectedRange` to accept a optional `fallbackRange` param. I'm not sure if you wanted this or to create a new function with `fallbackRange` param which returned the `DateRange` .